src/executor/mod.rs

```diff
@@ -0,0 +1,303 @@
+///
+/// # }
+/// ```
+pub fn global_sync_strategy() -> GlobalStrategy {
+    match COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY.get() {
+        Some(strategy) => GlobalStrategy::Initialized(strategy.into()),
+        None => GlobalStrategy::Default(<&SyncExecutor>::default().into()),
+    }
+}
+
+pub(crate) fn get_global_sync_executor() -> &'static SyncExecutor {
+    COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY
+        .get()
+        .unwrap_or_else(|| <&SyncExecutor>::default())
+}
+
+/// The stored strategy used to spawn compute-heavy futures.
+static COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY: OnceLock<SyncExecutor> = OnceLock::new();
+
+/// The fallback strategy used in case no strategy is explicitly set
+static DEFAULT_COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY: OnceLock<SyncExecutor> = OnceLock::new();
+
+#[non_exhaustive]
+pub(crate) enum SyncExecutor {
+    /// A non-op strategy that runs the function in the current context
+    CurrentContext(current_context::CurrentContextExecutor),
+    /// User-provided closure
+    CustomExecutor(custom_executor::CustomExecutor),
+    /// tokio task::spawn_blocking
+    #[cfg(feature = "tokio")]
+    SpawnBlocking(spawn_blocking::SpawnBlockingExecutor),
+}
+
+impl Default for &SyncExecutor {
+    fn default() -> Self {
+        DEFAULT_COMPUTE_HEAVY_SYNC_EXECUTOR_STRATEGY.get_or_init(|| {
+            log::warn!(
+                "Defaulting to CurrentContext (non-op) strategy for compute-heavy future executor"
+            );
+            SyncExecutor::CurrentContext(CurrentContextExecutor::new(None))
+        })
+    }
+}
+
+impl ExecuteSync for SyncExecutor {
+    async fn execute_sync<F, R>(&self, f: F) -> Result<R, Error>
+    where
+        F: FnOnce() -> R + Send + 'static,
+        R: Send + 'static,
+    {
+        match self {
+            SyncExecutor::CurrentContext(executor) => executor.execute_sync(f).await,
+            SyncExecutor::CustomExecutor(executor) => executor.execute_sync(f).await,
+            #[cfg(feature = "tokio")]
+            SyncExecutor::SpawnBlocking(executor) => executor.execute_sync(f).await,
+        }
+    }
+}
+
+impl From<&SyncExecutor> for ExecutorStrategy {
+    fn from(value: &SyncExecutor) -> Self {
+        match value {
+            SyncExecutor::CurrentContext(_) => Self::CurrentContext,
+            SyncExecutor::CustomExecutor(_) => Self::CustomExecutor,
+            #[cfg(feature = "tokio")]
+            SyncExecutor::SpawnBlocking(_) => Self::SpawnBlocking,
+        }
+    }
+}
+
+/// Initialize a set of sensible defaults for a tokio runtime:
+///
+/// - SpawnBlocking strategy
+/// - Max concurrency equal to the cpu core count.
+///
+/// # Error
+/// Returns an error if the global strategy is already initialized.
+/// It can only be initialized once.
+/// # Examples
+///
+/// ```
+/// use vacation::initialize_tokio;
+///
+/// # fn run() {
+/// initialize_tokio().unwrap();
+/// # }
+pub fn initialize_tokio() -> Result<(), Error> {
```

Comment:
the name of this function sounds like we're initializing the tokio runtime itself. I don't really have a better suggestion at this time though

Reply:
I mean, i can make it `initialize_tokio_strategy()`